9.0/9.0.1 Commerce/download-and-extract.ps1

```diff
@@ -0,0 +1,206 @@
+    if (Test-Path ".\\$file") {
+        $cred = Import-Clixml ".\\$file"
+    }
+    else {
+        $cred = Get-Credential -Message "Enter your SDN download credentials:"
+        $cred | Export-Clixml ".\\$file"
+    }
+ 
+    return $cred
+}
+
+Function Invoke-FetchDownloadAuthentication($cred) {
+    #   Credit: https://jermdavis.wordpress.com/2017/11/27/downloading-stuff-from-dev-sitecore-net/
+
+    $authUrl = "https://dev.sitecore.net/api/authorization"
+ 
+    $pwd = $cred.GetNetworkCredential().Password
+ 
+    $postParams = "{ ""username"":""$($cred.UserName)"", ""password"":""$pwd"" }"
+ 
+    $authResponse = Invoke-WebRequest -Uri $authUrl -Method Post -ContentType "application/json;charset=UTF-8" -Body $postParams -SessionVariable webSession
+    $authCookies = $webSession.Cookies.GetCookies("https://sitecore.net")
+ 
+    $marketPlaceCookie = $authCookies["marketplace_login"]
+ 
+    if ([String]::IsNullOrWhiteSpace($marketPlaceCookie)) {
+        throw "Credentials appear invalid"
+    }
+ 
+    $devUrl = "https://dev.sitecore.net"
+ 
+    $devResponse = Invoke-WebRequest -Uri $devUrl -WebSession $webSession
+    $devCookies = $webSession.Cookies.GetCookies("https://dev.sitecore.net")
+ 
+    $sessionCookie = $devCookies["ASP.Net_SessionId"]
+ 
+    return "$marketPlaceCookie; $sessionCookie"
+}
+
+function Invoke-SitecoreFileDownload {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [string] $Uri,
+ 
+        [Parameter(Mandatory)]
+        [string] $OutputFile,
+ 
+        [string] $cookies
+    )
+    #   Credit: https://jermdavis.wordpress.com/2017/11/27/downloading-stuff-from-dev-sitecore-net/
+
+    $webClient = New-Object System.Net.WebClient
+ 
+    if (!([String]::IsNullOrWhiteSpace($cookies))) {
+        $webClient.Headers.Add([System.Net.HttpRequestHeader]::Cookie, $cookie)
+    }
+ 
+    $data = New-Object psobject -Property @{Uri = $Uri; OutputFile = $OutputFile}
+ 
+    $changed = Register-ObjectEvent -InputObject $webClient -EventName DownloadProgressChanged -MessageData $data -Action {
+        Write-Progress -Activity "Downloading $($event.MessageData.Uri)" -Status "To $($event.MessageData.OutputFile)" -PercentComplete $eventArgs.ProgressPercentage
+    }
+ 
+    try {
+        $handle = $webClient.DownloadFileAsync($Uri, $PSCmdlet.GetUnresolvedProviderPathFromPSPath($OutputFile))
+ 
+        while ($webClient.IsBusy) {
+            Start-Sleep -Milliseconds 1000
+        }
+    }
+    finally {
+        Write-Progress -Activity "Downloading $Uri" -Completed
+ 
+        Remove-Job $changed -Force
+        Get-EventSubscriber | Where SourceObject -eq $webClient | Unregister-Event -Force
+    }    
+}
+Function Invoke-SitecoreDownload {
+    param(
+        [string]$url,
+        [string]$target
+    )
+ 
+    $cred = Invoke-FetchSitecoreCredentials
+    $cookie = Invoke-FetchDownloadAuthentication $cred
+ 
+    Invoke-SitecoreFileDownload -Uri $url -OutputFile $target -Cookies $cookie
```

Comment:
This all `Invoke-FetchSitecoreCredentials`, `Invoke-FetchDownloadAuthentication`, `Invoke-SitecoreFileDownload`
can be replaced with:
``` powershell
$loginRequest = Invoke-RestMethod -Uri https://dev.sitecore.net/api/authorization -Method Post -ContentType "application/json" -Body "{username: '$user', password: '$password'}" -SessionVariable session -UseBasicParsing
Invoke-WebRequest -Uri $package_url -WebSession $session -OutFile $package_zip_path -UseBasicParsing
```